wii-u-gc-adapter.c

```diff
@@ -275,11 +275,29 @@ static void update_ff_start_stop(struct ff_event *e, struct timespec *current_ti
 
 static int create_ff_event(struct ports *port, struct uinput_ff_upload *upload)
 {
+   bool stop = false;
+   switch (upload->effect.type)
+   {
+      case FF_PERIODIC:
+         stop = upload->effect.u.periodic.magnitude == 0;
+         break;
+      case FF_RUMBLE:
+         stop = upload->effect.u.rumble.strong_magnitude == 0 && upload->effect.u.rumble.weak_magnitude == 0;
+         break;
+   }
    if (upload->old.type != 0)
    {
       // events with length 0 last forever
```

Comment:
This comment should be moved to be above the `forever = (upload->effect.replay.length == 0)` line